kratos/utilities/mortar_utilities.h

```diff
@@ -359,6 +359,31 @@ class MortarUtilities
     }
     
     /**
+     * This function calculates the normal in a specific GP with a given shape function
+     * @param N: The shape function considered
+     * @param Geom: The geometry of condition of interest
+     */
+
+    static inline array_1d<double,3> GaussPointNormal(
+        const Vector& N,
+        const GeometryType& Geom
+        )
+    {
+        array_1d<double,3> normal = ZeroVector(3);
+        for( unsigned int i_node = 0; i_node < Geom.PointsNumber(); ++i_node )
+        {
+            normal += N[i_node] * Geom[i_node].GetValue(NORMAL); 
+        }
+        
+        if (norm_2(normal) > std::numeric_limits<double>::epsilon())
```

Comment:
the sum of non-coplanar unit vectors is not necessarily a unit vector.

also, instead of skipping the case silently,  i would throw an error if the norm is 0, at least in debug mode

Reply:
Ok